tests/Silverback.Integration.Tests.E2E/Kafka/ErrorHandlingTests.cs

```diff
@@ -1177,5 +1265,101 @@ public async Task MovePolicy_ToOtherTopicAfterRetry_MessageRetriedAndMoved()
             otherTopic.GetAllMessages()[0].Value.Should()
                 .BeEquivalentTo(Helper.Spy.InboundEnvelopes[0].RawMessage.ReReadAll());
         }
+
+        [Fact]
+        public async Task MovePolicy_ToOtherTopic_HeadersSet()
+        {
+            var message1 = new TestEventOne
+            {
+                Content = "Hello E2E msg1."
+            };
+            var message2 = new TestEventOne
+            {
+                Content = "Hello E2E msg2."
+            };
+            var message3 = new TestEventOne
+            {
+                Content = "Hello E2E msg3."
+            };
+
+            Host.ConfigureServices(
+                    services => services
+                        .AddLogging()
+                        .AddSilverback()
+                        .UseModel()
+                        .WithConnectionToMessageBroker(options => options.AddMockedKafka(
+                            mockOptions => mockOptions.WithDefaultPartitionsCount(1)))
```

Comment:
Shouldn't you try to produce to multiple partitions and see that the partitions header is correctly set?